datadog_checks_dev/datadog_checks/dev/tooling/e2e/docker.py

```diff
@@ -49,17 +50,35 @@ def agent_command(self):
             get_agent_exe(self.agent_version)
         )
 
-    def run_check(self, capture=False, rate=False):
-        command = '{} check {}{}'.format(
-            self.agent_command,
+    def exec_command(self, interactive=False):
+        return 'docker exec{} {}'.format(
+            ' -it' if interactive else '', self.container_name
+        )
+
+    def run_check(self, interactive=False, capture=False, rate=False):
+        command = '{} {} check {}{}'.format(
+            self.exec_command(interactive=interactive),
+            get_agent_exe(self.agent_version),
             self.check,
             ' {}'.format(get_rate_flag(self.agent_version)) if rate else ''
         )
+
         return run_command(command, capture=capture)
 
     def exists(self):
         return env_exists(self.check, self.env)
 
+    @contextmanager
+    def use_config(self, config):
```

Comment:
I do not understand this method very well. It seems it adds the config params missing in the config that will be used in the memory. Could you add some documentation and comment to make it easier to understand?